opm/simulators/wells/WellGroupControls.cpp

```diff
@@ -549,6 +556,116 @@ getGroupProductionTargetRate(const Group& group,
     return scale;
 }
 
+template<class Scalar>
+std::pair<Scalar, Group::ProductionCMode> WellGroupControls<Scalar>::
+getAutoChokeGroupProductionTargetRate(const std::string& name,
+                                      const Group& group,
```

Comment:
What is the difference between this code and getGroupProductionTargetRate? Can this be refactored to avoid code duplication.